src/operations.rs

```diff
@@ -1,29 +1,190 @@
+    state.registers[destination_register] = state.memory[memory_index];
+    update_flags(destination_register, &mut state.registers);
 }
 
-fn update_flags(register: u16, registers: &mut [u16; 10]) {
-    if registers[Registers::from(register)] == 0 {
+/// Load a value from memory to the destination register. <br>
+/// The memory direction is given by the value inside the base register and the sign extended offset
+/// * Instruction: |OP_Code (0110)|DR (3)|BaseR (3)|Offset (6)|<br>
+///   When finished update flags
+pub(crate) fn load_register(instruction: u16, state: &mut State) {
+    let sign_extended_offset = sign_extend(instruction & 0x3F, 6);
+    let base_register = Registers::from(instruction >> 6 & 0x7);
+    let destination_register = Registers::from(instruction >> 9 & 0x7);
+    let memory_index =
+        u16::wrapping_add(state.registers[base_register], sign_extended_offset) as usize;
+    state.registers[destination_register] = state.memory[memory_index];
+    update_flags(destination_register, &mut state.registers);
+}
+
+/// Load a memory address into a register
+/// * Instruction: |OP_Code (1110)|DR (3)|Offset (9)|<br>
+///   When finished update flags
+pub(crate) fn load_effective_address(instruction: u16, state: &mut State) {
+    let sign_extended_offset = sign_extend(instruction & 0x1FF, 9);
+    let destination_register = Registers::from((instruction >> 9) & 0x7);
+    let address = u16::wrapping_add(state.registers[Registers::Rpc], sign_extended_offset);
+    state.registers[destination_register] = address;
+    update_flags(destination_register, &mut state.registers);
+}
+
+/// Calculate the bitwise complement of the source registry and save it in the destination registry
+/// * Instruction: |OP_Code (1001)|DR (3)|SR (3)|1|11111|<br>
+///   When finished update flags
+pub(crate) fn not(instruction: u16, state: &mut State) {
+    let source_registry = Registers::from((instruction >> 6) & 0x7);
+    let destination_registry = Registers::from((instruction >> 9) & 0x7);
+    state.registers[destination_registry] = !(state.registers[source_registry]);
+    update_flags(destination_registry, &mut state.registers);
+}
+
+/// Store the contents of a source register into a specific location in memory
+/// * Instruction: |OP_Code (0011)|SR (3)|PCOffset (9)|<br>
+pub(crate) fn store(instruction: u16, state: &mut State) {
+    let sign_extended_offset = sign_extend(instruction & 0x1FF, 9);
+    let source_register = Registers::from((instruction >> 9) & 0x7);
+    let memory_address =
+        u16::wrapping_add(state.registers[Registers::Rpc], sign_extended_offset) as usize;
+    state.memory[memory_address] = state.registers[source_register];
+}
+
+/// The instruction receives the address where the memory address for storing the value is located, it reads it and saves the register value in it
+/// * Instruction: |OP_Code (1011)|SR (3)|PCOffset (9)|<br>
```

Comment:
We could make this clearer by changing it to something like the following:
```suggestion
/// The instruction takes the memory address containing the memory location where the source register's value should be stored and stores it.
/// * Instruction: |OP_Code (1011)|SR (3)|PCOffset (9)|<br>
```